src/streamable/deserialize.rs

```diff
@@ -0,0 +1,519 @@
+use super::error::{Error, Result};
+use std::convert::TryInto;
+
+pub struct ChiaDeserializer<'storage> {
```

Comment:
I think modules like this are traditionally named `ser.rs`.

Reply:
As you know, I forked `bincode` and made some tweaks that let it support "chia" serialization (where `usize` is encoded as `u32` instead of `u64`). I think an ideal future would be getting it work a bit more nicely and contributing it back to `bincode` so we can use it instead of rolling our own. But for sure, this is fine for now. Maybe we put it in a separate module called `chia_bincode` or `bincode_chia` or `bincode_u32` or something?

Reply:
The reason I went for a from-scratch implementation is that `bincode` supports a *lot* of options (most of them aren't relevant to us), and is quite a lot of code. It's a lot easier (I think) to review and be confident in the simpler implementation I made, that it does in fact implement our protocol.

`bincode` also supports constructs that `Streamable` doesn't support, and I think it would be risky to allow the opportunity to serialize such constructs, into invalid `Streamable` format. My implementation fails with `NotSupported` instead. In an ideal world, it would be a compile-time error, but I can't think of a way to do that, since traits must be fully implemented, even parts that are never used.